src/shared/locked-status/use-check-lock-status.js

```diff
@@ -0,0 +1,84 @@
+import { useEffect } from 'react'
+import { getCurrentDate } from '../fixed-periods/index.js'
+import { useMetadata, selectors } from '../metadata/index.js'
+import {
+    usePeriodId,
+    useDataSetId,
+    useOrgUnitId,
+} from '../use-context-selection/use-context-selection.js'
+import { useDataValueSet } from '../use-data-value-set/use-data-value-set.js'
+import { LockedStates, BackendLockStatusMap } from './locked-states.js'
+import { useLockedContext } from './use-locked-context.js'
+
+export const useCheckLockStatus = () => {
+    const [dataSetId] = useDataSetId()
+    const [orgUnitId] = useOrgUnitId()
+    const [periodId] = usePeriodId()
+    const { data: metadata } = useMetadata()
+    const { setLockStatus } = useLockedContext()
+    const dataValueSet = useDataValueSet()
+
+    useEffect(() => {
+        const now = getCurrentDate()
+
+        if (dataSetId && periodId) {
+            const dataInputPeriod = selectors.getApplicableDataInputPeriod(
+                metadata,
+                dataSetId,
+                periodId
+            )
+
+            if (dataInputPeriod) {
+                const openingDate = new Date(dataInputPeriod.openingDate)
+                const closingDate = new Date(dataInputPeriod.closingDate)
+
+                if (openingDate > now || closingDate < now) {
+                    // mark as invalid for data input period
+                    setLockStatus(LockedStates.LOCKED_DATA_INPUT_PERIOD)
+                    return
+                }
+            }
+        }
```

Comment:
Less nesting, easier to parse (imo):

```suggestion
        // should be outside of the hook
        const isDataInputPeriodLocked = (dataSetId, periodId) => {
            const dataInputPeriod = selectors.getApplicableDataInputPeriod(
                metadata,
                dataSetId,
                periodId
            )
            
            if (!dataInputPeriod) {
                return false
            }
            
            const openingDate = new Date(dataInputPeriod.openingDate)
            const closingDate = new Date(dataInputPeriod.closingDate)
            const now = getCurrentDate()
            
            return openingDate > now || closingDate < now
        }

        if (isDataInputPeriodLocked(dataSetId, periodId)) {
            // mark as invalid for data input period
            setLockStatus(LockedStates.LOCKED_DATA_INPUT_PERIOD)
            return
        }
```

Reply:
updated